File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        ll_len = 0

        curr = head
        while curr:
            ll_len += 1
            curr = curr.next
        
        n = ll_len - n + 1 # position of the node to be removed
        print(n)
        if n == 1:
            return head.next

        i = 1
        curr = head
        while curr.next:
            if i + 1 == n:
                curr.next = curr.next.next
                break
            i += 1
            curr = curr.next
        
        return head

        



        

File: Data Structures & Algorithms/pow-x-n/submission-0.cpp
class Solution {
public:
    double myPow(double x, int n) {
        double out = n >= 0? x: 1/x;
        
        if (n == 0) {
            return 1;

        } else if (n >= 1) {
            n--;
            while (n) {
                out *= x;
                n--;
            }

        } else {
            n++;
            while (n) {
                out *= 1 / x;
                n++;
            }
        }

        return out;
    }
};
